### PR TITLE
DM-28580: Documentation on how to get started with nublado on the test-stands and in Chile

### PR DESCRIPTION
**Pull request by tribeiro**
Add documentation on how to get started with the observatory control tools, namely;

	- ArgoCD: for deployment.
	- Chronograf: Visualizing data in the EFD.
	- LOVE: LSST Operation Visualization Environment, contains GUIs for the observatory control system.
	- Nublado: Jupyter lab environment with observatory control capabilities.

### CODE REVIEW

File: Observing-Interface-Setup/nublado-notebook-environment-update.rst [hunk 1]
```diff
@@ -163,7 +167,7 @@ Skip down to the :ref:`summary section <Update-Notebook-Environment-in-Nublado-S
 Here is a summary of the user's command inputs described in the procedure above:
 
 .. code-block:: bash
-   
+
```
Comment (mareuter) [inline Observing-Interface-Setup/nublado-notebook-environment-update.rst:170]: Because of the way the changes were committed, its hard to actually comment on some things. In the above, step 3 will not work because you aren't in a git directory. After the clone you need to cd first. That's done as part of step 4, but needs to be done earlier.

**Review by mareuter — COMMENTED** [review]
The way the commits were made cause sections of certain documents to not be reviewable. Overall the structure is OK. I think an auth section needs to be added until some services switch over to LDAP. Timing may make this not worth the effort. What are we going to do about the large number of empty sections? Are we mainly reviewing structure at this point?

File: Observing-Interface-Setup/environments.rst [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+Summit
+------
+
+Summit is our main production environment.
+Systems running here will be directly controlling hardware or communicating with components that control actual hardware.
+
+  .. important::
+
+      In the case of the Summit it is required to have personnel present at the site prior to any activity that involves moving hardware.
+      You may also be required to issue a jira ticket in the `summit activity project <https://jira.lsstcorp.org/projects/SUMMIT>`__.
```
Comment (patrickingraham) [inline Observing-Interface-Setup/environments.rst:146]: So not just anyone is allowed to file tickets here at the moment (permissions wise you can, procedure wise you can't)... 
Maybe just say it has to be a planned activity and put into the "to-do" state by the personnel organizing summit activities?

File: Observing-Interface-Setup/environments.rst [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+Troubleshooting
+===============
+
+.. This section should include troubleshooting information. Information in this section should be strictly related to this procedure.
+
+.. If there is no content for this section, remove the indentation on the following line instead of deleting this sub-section.
+
+If you can not open the links to the environment you intend to work with, make sure you are connected to the LSST-WAP wifi network in one of the designed areas (Tucson, La Serena or Summit facilities) or that you are connected to the NOAO VPN.
+
+If problems persist, you can ask for help in the designed Slack channels or in the com-square channel.
```
Comment (patrickingraham) [inline Observing-Interface-Setup/environments.rst:165]: designed = designated ?

File: Observing-Interface-Setup/environments.rst [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+.. _Observing-Interface-Getting-Started-Personnel:
+
+Contact Personnel
+=================
+
+This procedure was last modified |today|.
+
+This procedure was written by |author|. The following are contributors: |contributors|.
```
Comment (patrickingraham) [inline Observing-Interface-Setup/environments.rst:174]: feel free to change this to make it more useful. No need to stick to the template if you don't want to.
you could imagine putting, "for questions involving interface connection problems, ask X. For issues involving LOVE, contact Y" or whatever you wish.

File: Observing-Interface-Setup/getting-started-argocd.rst [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+.. Each section should includes a label for cross referencing to a given area.
+.. Recommended format for all labels is "Title Name"-"Section Name" -- Spaces should be replaced by hyphens.
+.. To reference a label that isn't associated with an reST object such as a title or figure, you must include the link an explicit title using the syntax :ref:`link text <label-name>`.
+.. An error will alert you of identical labels during the build process.
+
+###########################
+Getting Started With ArgoCD
+###########################
+
+.. _Observing-Interface-Getting-Started-ArgoCD-Overview:
```
Comment (patrickingraham) [inline Observing-Interface-Setup/getting-started-argocd.rst:25]: Maybe just put a note here saying this is under construction and not yet populated?
Or maybe just keep it for a different PR when you can populate it?

File: Observing-Interface-Setup/getting-started-chronograf.rst [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+.. _Observing-Interface-Getting-Started-Chronograf:
+.. Each section should includes a label for cross referencing to a given area.
+.. Recommended format for all labels is "Title Name"-"Section Name" -- Spaces should be replaced by hyphens.
+.. To reference a label that isn't associated with an reST object such as a title or figure, you must include the link an explicit title using the syntax :ref:`link text <label-name>`.
+.. An error will alert you of identical labels during the build process.
+
+###############################
+Getting Started With Chronograf
+###############################
+
```
Comment (patrickingraham) [inline Observing-Interface-Setup/getting-started-chronograf.rst:24]: same comment as above (pasting here so you don't have to scroll up and get carpal tunnel).
Maybe just put a note here saying this is under construction and not yet populated?
Or maybe just keep it for a different PR when you can populate it?

File: Observing-Interface-Setup/getting-started-love.rst [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+#########################
+
+.. _Observing-Interface-Getting-Started-LOVE-Overview:
+
+Overview
+========
+
+.. This section should provide a brief, top-level description of the procedure's purpose and utilization. Consider including the expected user and when the procedure will be performed.
+
+This procedure is intended for users of the Rubin Observatory that are interested in interacting with the  Observatory Control System using LOVE.
```
Comment (patrickingraham) [inline Observing-Interface-Setup/getting-started-love.rst:32]: probably worth defining the acronym in the first sentence.
but then the same comment as above regarding the next PR.

File: Observing-Interface-Setup/getting-started-nublado.rst [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+.. important::
+
+    In the case of the :ref:`Summit <Observing-Interface-Environments-Summit>` it is required to have personnel present at the site prior to any activity that involves moving hardware.
+    You may also be required to issue a jira ticket in the `summit activity project <https://jira.lsstcorp.org/projects/SUMMIT>`__.
+
+Once you announced your intention to operate one of the environments in the appropriate Slack channel and, if appropriate, coordinated the activities for the :ref:`Summit <Observing-Interface-Environments-Summit>`, you are ready to open nublado on your browser.
+
+Once you open the link on your browser, you should see the :ref:`nublado welcome page <Observing-Interface-Getting-Started-Nublado-Welcome>`.
+To start your nublado instance, click in the icon below "Notebook Aspect".
```
Comment (patrickingraham) [inline Observing-Interface-Setup/getting-started-nublado.rst:84]: should Nublado always be capitalized? Unclear from square documentation...

File: Observing-Interface-Setup/getting-started-nublado.rst [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+To clone the `ts_notebooks`_ repository do;
+
+  .. prompt:: bash
+
+      cd ~/notebooks
+      git clone https://github.com/lsst-ts/ts_notebooks.git
+
+You can than navigate to the location where the repository was cloned using the file browser on the left hand side of the screen, clicking on "notebooks" and than "ts_notebooks".
+There are useful notebooks to start with in the "examples" and "procedures" folders.
+
```
Comment (patrickingraham) [inline Observing-Interface-Setup/getting-started-nublado.rst:150]: I think this is a good place to put a sentence about setting up environments and link to your other page.

File: Observing-Interface-Setup/nublado-notebook-environment-update.rst [hunk 10]
```diff
@@ -72,7 +74,8 @@ Post-Condition
 .. It is preferred to include them as a bulleted or enumerated list.
 .. Do not include actions in this section. Any action by the user should be included in the end of the Procedure section below. For example: Do not include "Verify the telescope azimuth is 0 degrees with the appropriate command." Instead, include this statement as the final step of the procedure, and include "Telescope is at 0 degrees." in the Post-condition section.
 
-- The requested software version(s) is deployed for use in the jupyter-lab notebook. This will over-ride the versions available on the server.
+- The requested software version(s) is deployed for use in the jupyter-lab notebook.
+  This will over-ride the versions available on the server.
```
Comment (patrickingraham) [inline Observing-Interface-Setup/nublado-notebook-environment-update.rst:78]: maybe mention it doesn't over-write them? Meaning they can just remove the setup lines etc and revert back to the base container packages?

**Review by patrickingraham — CHANGES_REQUESTED** [review]

File: Observing-Interface-Setup/environments.rst [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+========
+
+.. This section should provide a brief, top-level description of the procedure's purpose and utilization. Consider including the expected user and when the procedure will be performed.
+
+This procedure is intended for users of the Rubin Observatory that are interested in interacting with the  Observatory Control System.
+It contains basic information (with links and description) on how to connect to the observing interface tools at the summit and at the different test stands.
+
+- Each environment is composed of a `nublado`_ instance, an EFD instance (with Chronograf and other visualization tools) and LOVE interface.
+- Coordination on the use of each environment is managed through Slack channels.
+- Container deployment is orchestrated via ArgoCD and Kubernetes.
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply Observing-Interface-Setup/environments.rst:38]: I added a sentence saying that. I think further information would be better suited in an "environment specific" page.

File: Observing-Interface-Setup/getting-started-nublado.rst [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+    Screenshot of nublado server options page.
+
+In this page, you can now select the image and the resources you need for your work.
+
+  - The "image" selection specify which version of the science pipelines to load (the observatory control packages are always the same).
+
+  - The "size" option specify the amount of computing resources that will be made available for your jupyter-lab instance.
+
+In most cases, you will want to select the "Recommended" image and "Medium" size.
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply Observing-Interface-Setup/getting-started-nublado.rst:111]: I thought the idea is to use mostly medium and large if you are going to need more resources (like the phrase afterwards). Is that wrong? should we recommend "Large" for everyone?

File: Observing-Interface-Setup/getting-started-nublado.rst [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+    These are shared resources and, as such, require coordination to avoid interference with other activities.
+
+.. important::
+
+    In the case of the :ref:`Summit <Observing-Interface-Environments-Summit>` it is required to have personnel present at the site prior to any activity that involves moving hardware.
+    You may also be required to issue a jira ticket in the `summit activity project <https://jira.lsstcorp.org/projects/SUMMIT>`__.
+
+Once you announced your intention to operate one of the environments in the appropriate Slack channel and, if appropriate, coordinated the activities for the :ref:`Summit <Observing-Interface-Environments-Summit>`, you are ready to open nublado on your browser.
+
+Once you open the link on your browser, you should see the :ref:`nublado welcome page <Observing-Interface-Getting-Started-Nublado-Welcome>`.
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply Observing-Interface-Setup/getting-started-nublado.rst:83]: It seems like those instructions are more tailored towards the LSP (non-sal) than the ones for our environments. 

For instance it mentions `NCSA VPN` and kerberos. They also stop one step short.

File: Observing-Interface-Setup/environments.rst [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+Overview
+========
+
+.. This section should provide a brief, top-level description of the procedure's purpose and utilization. Consider including the expected user and when the procedure will be performed.
+
+This procedure is intended for users of the Rubin Observatory that are interested in interacting with the  Observatory Control System.
+It contains basic information (with links and description) on how to connect to the observing interface tools at the summit and at the different test stands.
+
+- Each environment is composed of a `nublado`_ instance, an EFD instance (with Chronograf and other visualization tools) and LOVE interface.
+- Coordination on the use of each environment is managed through Slack channels.
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply Observing-Interface-Setup/environments.rst:37]: uhm... that is a good point. I honestly don't have a good answer for that.

File: Observing-Interface-Setup/environments.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (tribeiro) [reply Observing-Interface-Setup/environments.rst:38]: Ok, I did some modification to this sentence.

File: Observing-Interface-Setup/environments.rst [hunk 15]
```diff
@@ -0,0 +1,174 @@
...
+    - You will be prompted for your password twice after running the command above.
+    - Once logged in, open http://localhost:8080/ on a browser.
+
+- ArgoCD: https://lsst-argocd-nts-efd.ncsa.illinois.edu/
+- Slack: ncsa-integ-teststand
+
+
+.. _Observing-Interface-Environments-BTS:
+
+Base Test Stand (BTS)
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply Observing-Interface-Setup/environments.rst:106]: changed it to "Base Facility Test Stand"... should we name it "BFTS"?

File: Observing-Interface-Setup/environments.rst [hunk 16]
```diff
@@ -0,0 +1,174 @@
...
+---------------------
+
+BTS is a simulation environment running at the base facility in La Serena.
+The main difference between the :ref:`NTS <Observing-Interface-Environments-NTS>` and this environment is that some systems will run using hardware simulators.
+
+Hardware simulators are, in general, higher fidelity than those of pure software available at :ref:`NTS <Observing-Interface-Environments-NTS>`.
+For that, BTS will be used mostly for higher fidelity testing, development and debugging.
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply Observing-Interface-Setup/environments.rst:113]: maybe?

File: Observing-Interface-Setup/environments.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (patrickingraham) [reply Observing-Interface-Setup/environments.rst:37]: how about "operational environment" for summit/base/nts ?

File: Observing-Interface-Setup/environments.rst [hunk 17]
```diff
@@ -0,0 +1,185 @@
...
+.. _DM-28465: https://jira.lsstcorp.org/browse/DM-28465
+.. _ssh with Kerberos: https://developer.lsst.io/services/lsst-login.html?highlight=kerberos#ssh-with-kerberos
+
+.. _Observing-Interface-Environments-BTS:
+
+Base Facility Test Stand (BTS)
+------------------------------
+
+BTS is a simulation environment running at the base facility in La Serena.
+The main difference between the :ref:`NTS <Observing-Interface-Environments-NTS>` and this environment is that some systems will run using hardware simulators.
```
Comment (patrickingraham) [inline Observing-Interface-Setup/environments.rst:121]: and this operational environment --- but only if you want to adopt that

File: Observing-Interface-Setup/environments.rst (hunk 16 above)
[the remark this answers is not in the document]
Comment (patrickingraham) [reply Observing-Interface-Setup/environments.rst:124]: Maybe just add a sentence saying that this teststand will be particularily useful for testing of lower-level hardware interfaces because of the presence of the simulators.

File: Observing-Interface-Setup/getting-started-nublado.rst (hunk 12 above)
[the remark this answers is not in the document]
Comment (patrickingraham) [reply Observing-Interface-Setup/getting-started-nublado.rst:111]: I don't think people will know what the definition of a "computationally intensive task" is.
I'd say something like, "for anyone performing testing a Large container should be used as it requires significant resources. A small/medium container should only be used when performing small tasks, such as updating notebooks or files."

**Review by patrickingraham — APPROVED** [review]
I made a few minor comments. Please take a look and see if it makes sense to incorporate them before merging.
